### PR TITLE
refactoring , MACRO

### PR DESCRIPTION
**Pull request by PavelAtCarbyne**

### CODE REVIEW

File: plugins/janus_videoroom.c [hunk 1]
```diff
@@ -1488,6 +1488,28 @@ typedef enum  forward_media_type {
 				(MEDIA_VIDEO   == media_type ? "Video"  :              \
 				(MEDIA_AUDIO_MIXER == media_type ? "AudioMixer" : "unknown"))))
 
+#define RETURN_FALSE_IF_EQUAL_LOG_ERROR(value1, value2, format, ...) \
```
Comment (alex-dizengof) [inline plugins/janus_videoroom.c]: VERIFY_ELSE_RETURN_FALSE

in this style

#define VERIFY_ELSE_CLEANUP(condition, message)          \
	if(!(condition))                                     \
	{                                                    \
		GST_ERROR((message));                            \
		goto CLEANUP;                                    \
	}

File: plugins/janus_videoroom.c [hunk 2]
```diff
@@ -1488,6 +1488,28 @@ typedef enum  forward_media_type {
...
+do { \
+        if(value1 == value2) { \
+                JANUS_LOG(LOG_ERR,format,##__VA_ARGS__); \
+                return FALSE; \
+        } \
+} while (0)
+
+#define RETURN_VOID_IF_EQUAL_LOG_ERROR(value1, value2, format, ...) \
```
Comment (alex-dizengof) [inline plugins/janus_videoroom.c]: VERIFY_ELSE_RETURN

File: plugins/janus_videoroom.c [hunk 3]
```diff
@@ -1488,6 +1488,28 @@ typedef enum  forward_media_type {
...
+do { \
+        if(value1 == value2) { \
+                JANUS_LOG(LOG_ERR,format,##__VA_ARGS__); \
+                return; \
+        } \
+} while (0)
+
+#define GOTO_CLEANUP( format, ...) \
```
Comment (alex-dizengof) [inline plugins/janus_videoroom.c]: LOG_ERROR_AND_GOT_TO_CLEANUP

File: plugins/janus_videoroom.c [hunk 4]
```diff
@@ -1488,6 +1488,28 @@ typedef enum  forward_media_type {
...
+do { \
+	JANUS_LOG(LOG_ERR,format,##__VA_ARGS__); \
+        goto CLEANUP; \
+} while (0)
+
```
Comment (alex-dizengof) [inline plugins/janus_videoroom.c:1512]: Add also

VERIFY_ELSE_GOTO_CLEANUP

**Review by alex-dizengof — CHANGES_REQUESTED** [review]

File: plugins/janus_videoroom.c (hunk 4 above)
Comment (PavelAtCarbyne) [reply plugins/janus_videoroom.c:1512]: done

File: plugins/janus_videoroom.c (hunk 1 above)
Comment (PavelAtCarbyne) [reply plugins/janus_videoroom.c]: done

File: plugins/janus_videoroom.c (hunk 2 above)
Comment (PavelAtCarbyne) [reply plugins/janus_videoroom.c]: done

File: plugins/janus_videoroom.c (hunk 3 above)
Comment (PavelAtCarbyne) [reply plugins/janus_videoroom.c]: done

**Review by alex-dizengof — APPROVED** [review]

**Pull request merged**